newIDE/app/src/SceneEditor/InstancesSelection.js

```diff
@@ -24,21 +24,33 @@ export default class InstancesSelection {
     this.selection.length = 0;
   }
 
-  selectInstance(instance: gdInitialInstance, multiselect: boolean = false) {
+  selectInstance(
+    instance: gdInitialInstance,
+    multiselect: boolean = false,
+    layersVisibility: object
+  ) {
     if (this.isInstanceSelected(instance)) {
       if (multiselect) this.unselectInstance(instance);
 
       return;
     }
 
     if (!multiselect) this.clearSelection();
-    this.selection.push(instance);
+    if (layersVisibility[instance.getLayer()]) {
```

Comment:
Let's make layersVisibility optional by allowing null/undefined to be passed. This will be useful because `selectInstance` is used in SceneEditor/index.js for selecting instances from the instances list (_onSelectInstances). In this case, no layer visibility should be used, so we should pass null/undefined and the function should still work..

Reply:
good thinking, that way I dont have to feed the layervisibility object to it from instances editor - where we still want to be able to select instances on hidden layers